core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -219,7 +219,7 @@ class BaseUnit : INamed, INonPerpetualConstruction, ICivilopediaText {
         ignoreCityRequirements: Boolean
     ): Boolean {
         // May buy [unitFilter] units for [amount] [Stat] starting from the [eraName] at an increasing price ([amount])
-        if (cityInfo.civInfo.getMatchingUniques("May buy [] units for [] [] [] starting from the [] at an increasing price ([])")
+        if (!ignoreCityRequirements && cityInfo.civInfo.getMatchingUniques("May buy [] units for [] [] [] starting from the [] at an increasing price ([])")
```

Comment:
You seem to have 3 [] instead of 2?

Reply:
ah, missed the cityFilter in the comment, I'll add it in